src/queryEngine.ts

```diff
@@ -160,47 +161,68 @@ export async function queryEngine(): Promise<void> {
   }
 }
 
+const filterAddNewTxs = async (
+  pluginId: string,
+  dbTransactions: nano.DocumentScope<StandardTx>,
+  docIds: string[],
+  transactions: StandardTx[]
+): Promise<void> => {
+  if (docIds.length < 1 || transactions.length < 1) return
+  const queryResults = await dbTransactions.fetch(
+    { keys: docIds },
+    { include_docs: false }
+  )
+
+  const newDocs: DbTx[] = []
+  for (const docId of docIds) {
+    if (
+      queryResults.rows.find(doc => 'id' in doc && doc.id === docId) == null
+    ) {
```

Comment:
The code previously threw query errors with the exception of 'not_found' errors. Looks like the new implementation is silent when it comes to rows which returned an error. Is this intentional?

Reply:
The only kind of errors other than `not_found` would be a network error accessing the db. That would still throw since the entire `dbTransactions.fetch` would throw. The loop on line 177 is basically a way of checking `not_found`

Reply:
If I'm not mistaken each row's type is `DocumentResponseRow<D> | DocumentLookupFailure` and it's the `DocumentLookupFailure` case we're being silent about. It'll contain an error string for a particular document key, such as `"not_found"`, but I wouldn't assume this to be the only error string that may be returned from fetch; couchdb I'm sure has other error strings which it may return for the document retrieval. We may be fine ignoring these errors, but perhaps at least logging them if there were to occur improves our system observability.